src/DotNet/DotNet.csproj

```diff
@@ -97,11 +98,39 @@
     />
   </Target>
 
+  <!-- if we are building in the unified pipeline we want to install the runtimes from the props of the sdks -->
+  <Target Name="_GetExternalDotNetSdkVersions">
+    <XmlPeek
+        Condition="Exists('$(AndroidSrcPath)\eng\Versions.props')"
+        XmlInputPath="$(AndroidSrcPath)\eng\Versions.props"
+        Query="/Project/PropertyGroup/MicrosoftNETCoreAppRefPackageVersion/text()">
+      <Output TaskParameter="Result" PropertyName="VersionFromAndroid" />
+    </XmlPeek>
+    <XmlPeek
+        Condition="Exists('$(MaciosSrcPath)\eng\Versions.props')"
+        XmlInputPath="$(MaciosSrcPath)\eng\Versions.props"
+        Query="/Project/PropertyGroup/MicrosoftNETCoreAppRefPackageVersion/text()">
+      <Output TaskParameter="Result" PropertyName="VersionFromMacios" />
+    </XmlPeek>
+    <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('windows'))">
+      <DotNetXARuntimeInstallCommand>&amp; '$(DotNetInstallScriptPath)' -Version $(VersionFromAndroid) -InstallDir '$(DotNetDirectory)' -Verbose -Runtime dotnet</DotNetXARuntimeInstallCommand>
```

Comment:
This will install a new runtime into the .NET SDK. So, you get a folder like `dotnet/host/fxr/7.0.*`.

This allows `dotnet build` to run with a newer runtime, but I don't think it will actually provision a newer Mono workload. So on-device tests & things will have the older one?

Do we need to figure this out as well?

Reply:
The main reasoning behind this is to fix the MAUI build in the megapipeline. If Android builds with an sdk that references a newer runtime than the one that MAUI provisions, tools like class-parse won't run:
```
Using: D:\a\_work\1\s\bin\dotnet\dotnet.exe D:\a\_work\1\s\bin\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.30\tools\class-parse.dll
    [class-parse] response file: obj\Release\net7.0-android\class-parse.rsp
    --o="obj\Release\net7.0-android\api.xml.class-parse"
    "obj\Release\net7.0-android\library_project_jars\maui.aar\classes.jar"
    CommandLineArguments = D:\a\_work\1\s\bin\dotnet\dotnet.exe D:\a\_work\1\s\bin\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.30\tools\class-parse.dll "@obj\Release\net7.0-android\class-parse.rsp" 
    You must install or update .NET to run this application.
    App: D:\a\_work\1\s\bin\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.30\tools\class-parse.dll
    Architecture: x64
    Framework: 'Microsoft.NETCore.App', version '7.0.0' (x64)
    .NET location: D:\a\_work\1\s\bin\dotnet\
    The following frameworks were found:
      7.0.0-rtm.22511.4 at [D:\a\_work\1\s\bin\dotnet\shared\Microsoft.NETCore.App]
```
 I think we may have a similar situation with some of the tools that ship with XM/XI SDKs as well. Ideally the megapipeline will add support to build everything against the same .NET SDK dependency, but that is not currently the case.